cpp/oneapi/dal/algo/pca/backend/cpu/train_kernel_cov.cpp

```diff
@@ -44,13 +45,12 @@ static result_t call_daal_kernel(const context_cpu& ctx,
                                  const table& data) {
     const std::int64_t column_count = data.get_column_count();
     const std::int64_t component_count = get_component_count(desc, data);
-
+    auto result = train_result<task_t>{}.set_result_options(desc.get_result_options());
```

Comment:
It would be good to check that column_count > 0 and component_count > 0 - just to avoid crash in array::empty called